lib/src/runner/configuration.dart

```diff
@@ -605,4 +450,26 @@ class Configuration {
           Map<Object, Configuration> map2) =>
       mergeMaps(map1, map2,
           value: (config1, config2) => config1.merge(config2));
+
+  /// Returns a copy of this [Configuration] with all [chosenPresets] resolved
+  /// against [presets].
+  Configuration _resolvePresets() {
+    if (chosenPresets.isEmpty || presets.isEmpty) return this;
+
+    var newPresets = new Map<String, Configuration>.from(presets);
+    var merged = chosenPresets.fold(empty, (merged, preset) {
+      if (!newPresets.containsKey(preset)) return merged;
```

Comment:
All small tweaks are worth it if they're good. I can even make it cleaner, since `Configuration.merge(Configuration.empty)` is a no-op. Done.